ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java

```diff
@@ -2805,7 +2793,7 @@ public void testDynPartUpdateWithAborts() throws Exception {
     hiveConf.setBoolVar(HiveConf.ConfVars.HIVE_TEST_MODE_FAIL_LOAD_DYNAMIC_PARTITION, true);
     runStatementOnDriverWithAbort("update " + Table.ACIDTBLPART + " set b=a+2 where a<5");
     hiveConf.setBoolVar(HiveConf.ConfVars.HIVE_TEST_MODE_FAIL_LOAD_DYNAMIC_PARTITION, false);
-    verifyDeltaDirAndResult(2, Table.ACIDTBLPART.toString(), "p=p1", resultData1);
+    verifyDeltaDirAndResult(3, Table.ACIDTBLPART.toString(), "p=p1", resultData1);
```

Comment:
why do we have changes of expected results after the engine change?

Reply:
In this test we make the update statement fail explicitly by setting `HIVE_TEST_MODE_FAIL_LOAD_DYNAMIC_PARTITION` to `true`.
With Tez the jobs has a chance to finish their work and both the delta and delete delta directories and buckets are written hence we end up with 3 deltas and 1 delete delta.
With MR the insert job fails before creating the delta directory so we end up with 2 deltas only.
The txn is marked failed and rolled back with both execution engines.

Reply:
ok